File: src/createTemplate.js
import { readdirSync, writeFileSync } from "fs";
import path from "path";

const templateDir = "./src/templates";
const templates = readdirSync(templateDir).filter((file) =>
  readdirSync(path.join(templateDir, file)).includes("schema.json"),
);

writeFileSync(
  "./src/static/templates.json",
  JSON.stringify(templates, null, 2),
);

File: createTemplate.py
import os
import json

template_dir = "./src/templates"
templates = [
    folder for folder in os.listdir(template_dir)
    if "schema.json" in os.listdir(os.path.join(template_dir, folder))
]

with open("./src/static/templates.json", "w") as file:
    json.dump(templates, file, indent=2)
